build-tools/packages/build-cli/src/commands/check/policy.ts

```diff
@@ -0,0 +1,243 @@
+            dur += Date.now() - start;
+
+            actionMap.set(name, dur);
+            handlerActionPerf.set(action, actionMap);
+            return result;
+        };
+
+        // route files to their handlers by regex testing their full paths
+        // synchronize output, exit code, and resolve decision for all handlers
+        const routeToHandlers = (file: string) => {
+            handlers
+                .filter((handler) => handler.match.test(file) && handlerRegex.test(handler.name))
+                .map((handler) => {
+                    const result = runWithPerf(handler.name, "handle", () =>
+                        handler.handler(file, pathToGitRoot),
+                    );
+                    if (result) {
+                        let output = `${newline}file failed policy check: ${file}${newline}${result}`;
+                        const resolver = handler.resolver;
+                        if (flags.resolve && resolver) {
+                            output += `${newline}attempting to resolve: ${file}`;
+                            const resolveResult = runWithPerf(handler.name, "resolve", () =>
+                                resolver(file, pathToGitRoot),
+                            );
+
+                            if (resolveResult.message) {
+                                output += newline + resolveResult.message;
+                            }
+
+                            if (!resolveResult.resolved) {
+                                this.exit(1);
+                            }
+                        } else {
+                            this.exit(1);
+                        }
+
+                        this.log(output);
+                    }
+                });
+        };
+
+        const handleLine = (line: string) => {
+            const filePath = path.join(pathToGitRoot, line).trim().replace(/\\/g, "/");
+
+            if (pathRegex.test(line) && fs.existsSync(filePath)) {
+                count++;
+                if (exclusions.every((value) => !value.test(line))) {
+                    try {
+                        routeToHandlers(filePath);
+                    } finally {
+                        runPolicyCheck();
+                        logStats();
+                    }
+
+                    processed++;
+                } else {
+                    this.log(`Excluded: ${line}`);
+                }
+            }
+        };
+
+        const runPolicyCheck = () => {
+            for (const h of handlers) {
+                const final = h.final;
+                if (final) {
+                    const result = runWithPerf(h.name, "final", () =>
+                        final(pathToGitRoot, flags.resolve),
+                    );
+                    if (result?.error) {
+                        this.exit(1);
+                        this.log(result.error);
+                    }
+                }
+            }
+        };
+
+        const logStats = () => {
```

Comment:
Similar to runPolicyCheck -- this can be a pure function outside of the class that takes handlerActionPerf as input.